test/test_Quaternion.jl

```diff
@@ -227,35 +227,35 @@ end
         end
 
         @testset "exp" begin
-            @test exp(Quaternion(0, 0, 0, 0)) == Quaternion(1, 0, 0, 0, true)
-            @test exp(Quaternion(2, 0, 0, 0)) == Quaternion(exp(2), 0, 0, 0, false)
-            @test exp(Quaternion(0, 2, 0, 0)) == Quaternion(cos(2), sin(2), 0, 0, true)
-            @test exp(Quaternion(0, 0, 2, 0)) == Quaternion(cos(2), 0, sin(2), 0, true)
-            @test exp(Quaternion(0, 0, 0, 2)) == Quaternion(cos(2), 0, 0, sin(2), true)
-
+            @test exp(Quaternion(0, 0, 0, 0)) === Quaternion(1., 0., 0., 0., true)
+            @test exp(Quaternion(2, 0, 0, 0)) === Quaternion(exp(2), 0, 0, 0, false)
+            @test exp(Quaternion(0, 2, 0, 0)) === Quaternion(cos(2), sin(2), 0, 0, true)
+            @test exp(Quaternion(0, 0, 2, 0)) === Quaternion(cos(2), 0, sin(2), 0, true)
+            @test exp(Quaternion(0, 0, 0, 2)) === Quaternion(cos(2), 0, 0, sin(2), true)
+        
```

Comment:
Which spaces? The newlines? These are present on master.

I agree that it would have been better to have a formatter when the package was started (though none existed then), but I'd rather not add one now. Reformatting the entire package makes git blames to track the history for a bit of code useless.

Reply:
8 white spaces are added in line 235.

![image](https://user-images.githubusercontent.com/7488140/155296209-cd42d488-fb75-43a6-a965-a89a04d073cf.png)

We don't have the spaces in the current master branch.
https://github.com/JuliaGeometry/Quaternions.jl/blob/766add41cb246faed9fa6d403b2aace86a3303bb/test/test_Quaternion.jl#L250-L252

There also are 8 white spaces in lines 241, 247, 253, and 258.

![image](https://user-images.githubusercontent.com/7488140/155297061-1d1e2860-e1a0-4d0e-8a76-cae1e15dbdd3.png)

Reply:
>I'd rather not add one now.

I agree with that. It should not be in this PR.

Can I add an action for formatting with JuliaFormatter.jl? If this sounds good, I'll make a PR.
cf. https://github.com/julia-actions/julia-format/blob/master/workflows/format_check.yml

Note that the current master branch has 2 and 4 white spaces for indent, and that makes confusing for new contributers.
Formatting code may break git diff, but I think it should be done as soon as possible.

Reply:
I'd say open a PR and let's see how it looks. My concern is that, especially with the indentation, the formatter will touch almost every line of code, which would not be ideal. I'm all for clean, consistent formatting, but being able to explore the history easily is also important.